proxy_ring_buffer.go

```diff
@@ -0,0 +1,278 @@
+/*
+ * Copyright (c) 2008-2022, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package hazelcast
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	hzcerrors "github.com/hazelcast/hazelcast-go-client/internal/hzerrors"
+	"github.com/hazelcast/hazelcast-go-client/internal/proto/codec"
+	iserialization "github.com/hazelcast/hazelcast-go-client/internal/serialization"
+)
+
+// A Ringbuffer is a data structure where the content is stored in a ring-like
+// structure. A Ringbuffer has a fixed capacity, so it won't grow beyond
+// that capacity and endanger the stability of the system. If that capacity
+// is exceeded, the oldest item in the Ringbuffer is overwritten.
+//
+// For details see https://docs.hazelcast.com/hazelcast/latest/data-structures/ringbuffer
+type Ringbuffer struct {
+	*proxy
+	partitionID int32
+	capacity    int64
+}
+
+// OverflowPolicy
+// Using this OverflowPolicy one can control the behavior what should to be done
+// when an item is about to be added to the RingBuffer, but there is {@code 0}
```

Comment:
Will this {@code 0} work with godoc?  Also make `RingBuffer`s Ringbuffer for consistency. There are 3 of them in this file. Also there are 3 `ringbuffer`s

Reply:
Made consisten naming Ringbuffer.
ef100407591a499e28efa1f4619f5b9cd61fee6c